pkg/cloudprovider/providers/azure/azure_storage.go

```diff
@@ -19,23 +19,53 @@ package azure
 import (
 	"fmt"
 
+	"github.com/Azure/azure-sdk-for-go/arm/storage"
+	"github.com/Azure/go-autorest/autorest/to"
 	"github.com/golang/glog"
 )
 
+const (
+	defaultStorageAccountType  = string(storage.StandardLRS)
+	fileShareAccountNamePrefix = "f"
+)
+
 // CreateFileShare creates a file share, using a matching storage account
 func (az *Cloud) CreateFileShare(name, storageAccount, storageType, location string, requestGB int) (string, string, error) {
 	var err error
 	accounts := []accountWithLocation{}
-	if len(storageAccount) > 0 {
-		accounts = append(accounts, accountWithLocation{Name: storageAccount})
-	} else {
-		// find a storage account
-		accounts, err = az.getStorageAccounts()
+	if len(storageAccount) == 0 {
+		if location == "" {
+			location = az.controllerCommon.location
+		}
+		if storageType == "" {
+			storageType = defaultStorageAccountType
+		}
+		// find a storage account that matches storageType
+		accounts, err = az.getStorageAccounts(storageType)
```

Comment:
For windows cluster, a file service could lie in another region now.
Seems another case could happen, since we do not pass location here:
If the resource group is in locationA, and we have some storage account in locationA.
If user has specified locationB for file service spec, then here we are still getting some accounts ,thus will not create a new one.

But later in Line#71, it will do location check, but no storage account matches that, then an error would still be triggered.

Maybe we can merge the account check logic in Line#71 and do create if we could not find one matching.